src/Compilers/Core/Portable/PEWriter/MetadataWriter.cs

```diff
@@ -2117,33 +2110,19 @@ private void AddCustomAttributesToTable<T>(IEnumerable<T> parentList, Func<T, En
             foreach (var parent in parentList)
             {
                 EntityHandle parentHandle = getDefinitionHandle(parent);
-                foreach (ICustomAttribute customAttribute in parent.GetAttributes(Context))
-                {
-                    AddCustomAttributeToTable(parentHandle, customAttribute);
-                }
+                AddCustomAttributesToTable(parentHandle, parent.GetAttributes(Context));
             }
         }
 
-        private void AddCustomAttributesToTable(
-            EntityHandle handle,
-            ImmutableArray<ICustomAttribute> attributes)
+        protected virtual int AddCustomAttributesToTable(EntityHandle parentHandle, IEnumerable<ICustomAttribute> attributes)
```

Comment:
Changed this to IEnumerable so it could be used in more cases and only require the `DeltaMetadataWriter` to override one method. If its too much of a perf issue let me know, but the ImmutableArray overload was only called twice.